src/etcd/etcd_utils.h

```diff
@@ -0,0 +1,20 @@
+
+#ifndef CREDIS_UTILS_H
+#define CREDIS_UTILS_H
+
+#include <vector>
+#include "etcd3/include/etcd3.h"
+
+namespace utils {
+  class EtcdURL {
```

Comment:
How about using a `struct` instead to represent data containers?  You might be able to get rid of `public:` that way too